Packs/Automox/Integrations/Automox/Automox.py

```diff
@@ -0,0 +1,851 @@
+"""Automox Integration for Cortex XSOAR (aka Demisto)
+
+This integration empowers users of Cortex XSOAR with the
+capabilities provided by the Automox API.
+
+Organization, group, device, policy, and patch management all at your fingertips.
+
+For API reference, visit: https://developer.automox.com/
+"""
+
+import base64
+import string
+import traceback
+from typing import Any, Dict, List
+
+import demistomock as demisto
+import requests
+from CommonServerPython import *  # noqa # pylint: disable=unused-wildcard-import
+from CommonServerUserPython import *  # noqa
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()  # pylint: disable=no-member
+
+''' CONSTANTS '''
+VERSION = "1.0.0"
+USER_AGENT = f'ax:PaloAltoNetworks-XSOAR-plugin/{VERSION}'
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'  # ISO8601 format with UTC, default in XSOAR
+
+# Common keys used in command arguments
+ORG_IDENTIFIER = 'org_id'
+DEVICE_IDENTIFIER = 'device_id'
+GROUP_IDENTIFIER = 'group_id'
+LIMIT_IDENTIFIER = 'limit'
+PAGE_IDENTIFIER = 'page'
+
+DEFAULT_ORG_ID = demisto.params().get(ORG_IDENTIFIER, None)
+
+''' CLIENT CLASS '''
+class Client(BaseClient):
+    """Client class to interact with the service API
+
+    This Client implements API calls, and does not contain any XSOAR logic.
+    Should only do requests and return data.
+    It inherits from BaseClient defined in CommonServer Python.
+    Most calls use _http_request() that handles proxy, SSL verification, etc.
+    For this  implementation, no special attributes defined
+    """
+
+    def _get_list_results(self, method: string, url_suffix: string, params: dict) -> List[Dict]:
+        results = []
+
+        result_limit = int(params['limit'])
+        page_limit = 250 if int(params['limit']) > 250 else int(params['limit'])
+
+        params['limit'] = page_limit
+
+        while result_limit > 0:
```

Comment:
When is limit 0? Wouldn't this contradict with line 70 or what am I missing?

Reply:
I could have probably named this variable better. The users might specify a limit of 1000, when we go to make these requests, we chunk them into requests that have the max limit of 250. Each time, the result length is subtracted from the `result_limit` variable, so for example, after we paginate through 1000 records with 4 requests, `result_limit` should equal 0 and that's when we know to dip out of the loop.

Line 70 is there to short-circuit the loop in the event the user specified a limit larger than the total data set.

Reply:
Ah got it. I saw the break on 71 and for some reason saw it as a return statement instead.. So we are paging until we hit the limit and then returning. If response length is less than page_limit we break. Still not sure I get the `== 0` condition but understand the rest